### DISCUSSION TITLE
Ability to log time to an issue

### DISCUSSION THREAD

**Discussion by dmncr** [ankitpokhrel/jira-cli#181] [Ideas] [answered]
Could you please also add functionality so time spent can be logged to an issue? 

Thanks for putting in the effort to make jira usable ;)

**Comment by emahnovets** [comment]
Hey guys. Is there are any plans on getting this feature implemented in the foreseeable future? Personally, I'd like to get this option in cli since we are using time spent feature extensively

**Comment by ankitpokhrel** [comment]
Hi @dmncr @emahnovets, initial [support for adding a worklog ](https://github.com/ankitpokhrel/jira-cli#worklog=) is available in the main branch (#410). Please feel free to test the feature and report any issues. You can either [compile the source locally](https://github.com/ankitpokhrel/jira-cli#development) or use following command (requires go1.16+) to install the main build.

```sh
go install github.com/ankitpokhrel/jira-cli/cmd/jira@main
```

If you are using Homebrew, you can do:

```sh
brew install jira-cli --HEAD
```

**Reply by emahnovets** [reply]
Hi @ankitpokhrel . Just tested worklogs feature and it works fine 👍🏻 I there are a way to add ability to list worklogs for issue too?

**Comment by tardypad** [comment]
I'm moving away from `go-jira` to `jira-cli` for the automation of my Jira tasks which is mainly about adding worklogs.
So thanks a lot for the addition of this feature, it seems to work fine from my quick tests.
The thing I'm missing though now from my previous setup would be a **flag to specify the day of the worklog**
This is to allow adding missing worklogs in the past

**Reply by ankitpokhrel** [reply]
Hi @tardypad, this feature will be available in the new release. For now, If you wish to use this feature you can either [compile the source locally](https://github.com/ankitpokhrel/jira-cli#development) or if you are using homebrew you can install with `brew install jira-cli --HEAD` or you can use docker `docker run -it --rm ghcr.io/ankitpokhrel/jira-cli:main`.

Some examples:

```sh
# You can add a worklog with the specific start date (defaults to UTC timezone)
$ jira issue worklog add ISSUE-1 "2d 1h 30m" --started "2022-01-01 09:30:00"

# You can specify timezone to use along with the start date in IANA timezone format
$ jira issue worklog add ISSUE-1 3h --started "2022-01-01 09:30:00" --timezone "Europe/Berlin"

# Or, you can use start date in Jira datetime format and skip the timezone flag
$ jira issue worklog add ISSUE-1 "1h 30m" --started "2022-01-01T09:30:00.000+0200"
```

**Reply by tardypad** [reply]
Thanks a lot for adding this so fast!

**Comment by ankitpokhrel** [comment, marked as answer]
This feature is available since [v1.1.0](https://github.com/ankitpokhrel/jira-cli/releases/tag/v1.1.0). Latest release [v1.2.0](https://github.com/ankitpokhrel/jira-cli/releases/tag/v1.2.0) adds a flag `started` that allows you to specify a datetime on which the worklog effort was started.
